extension.js

```diff
@@ -312,12 +316,41 @@ function workarea_for_window(win) {
 }
 
 function target_rect_for_workarea(workarea) {
-    const target_rect = workarea.copy();
-    target_rect.height *= settings.get_double('window-height');
-    return target_rect;
+    const full_width = workarea.width + workarea.x,
+        full_height =  workarea.height + workarea.y,
+        width = workarea.width * settings.get_double('window-width'),
+        height = workarea.height * settings.get_double('window-height'),
+        horizontal_alignment = settings.get_string('window-horizontal-alignment'),
+        vertical_alignment = settings.get_string('window-vertical-alignment');
+    let x, y;
+    switch (horizontal_alignment) {
+    case 'left':
+        x = workarea.x;
+        break;
+    case 'right':
+        x = full_width - width;
+        break;
+    case 'center':
+        var x_diff = workarea.width  - width;
+        x = workarea.x + (x_diff === 0) ? 0 : Math.round(x_diff / 2);
```

Comment:
It doesn't look any simpler than the previous version. Why do you need the `if`/ternary operator? Isn't `0 / 2` equal to `0`?

Reply:
Done.